sdk/python/lib/pulumi/provider/experimental/analyzer.py

```diff
@@ -125,6 +125,9 @@ class Parameterization:
     # it in base64.
     value: str
 
+    def __hash__(self) -> int:
+        return hash((self.name, self.version, self.value))
```

Comment:
Guessing we need these classes to be mutable/we mutate them somewhere, and that's why `dataclass` won't generate `__hash__`? (Or rather, we don't set `frozen=True` in `dataclass`, so it won't generate `__hash__`)